lib/roadie/rails/asset_pipeline_provider.rb

```diff
@@ -12,11 +14,23 @@ def initialize(pipeline)
 
       def find_stylesheet(name)
         if (asset = find_asset_in_pipeline(name))
-          Stylesheet.new("#{asset.pathname} (live compiled)", asset.to_s)
+          Stylesheet.new("#{filename(asset)} (live compiled)", asset.to_s)
         end
       end
 
       private
+      def sprockets_4_or_later?
+        Sprockets::VERSION.split('.')[0].to_i > 3
+      end
+
+      def filename(asset)
+        if sprockets_4_or_later?
```

Comment:
I think in this case it's probably better.
In general i prefer version check as it's possible that old versions has this method but with different semantics, but in case of sprockets and filename we don't have this possibility.